ci/pipelines/Build.jenkinsfile

```diff
@@ -46,25 +46,23 @@ pipeline
 				{
-					def configurations = ['NewtonDebug', 'NewtonRelease', 'SnowDebug', 'SnowRelease']
-					for(int i=0; i < configurations.size(); i++)
-					{
-						stage('Build ' + configurations[i])
-						{
-							def props = readProperties file:"ci/pipelines/${configurations[i]}.properties"							
-							props.each { propKey, propValue -> println "${propKey}: ${propValue}" }
-
-							sh "mkdir build-${configurations[i]}"
-							dir("build-${configurations[i]}")
-							{
-								sh "conan install .. -s build_type=${props.conanBuildType} -s compiler.toolset=${props.conanCompilerToolset} -s arch=${props.conanArch} -o gtest=${props.conanGTestVersion} -o openssl=${props.conanOpenSSLVersion}"
-								sh "cmake .. -G '${props.cmakeVS}' -A ${props.cmakePlatform}"
-								sh "cmake --build . --config ${props.cmakeConfiguration}"
-								sh "conan export-pkg ../conanfile.py RESTAPICore/${version}@systelab/${channel} -s build_type=${props.conanBuildType} -s compiler.toolset=${props.conanCompilerToolset} -s arch=${props.conanArch} -o gtest=${props.conanGTestVersion} -o openssl=${props.conanOpenSSLVersion} --force"
-								sh "conan test ../test_package/conanfile.py RESTAPICore/${version}@systelab/${channel} -s build_type=${props.conanBuildType} -s compiler.toolset=${props.conanCompilerToolset} -s arch=${props.conanArch} -o gtest=${props.conanGTestVersion} -o openssl=${props.conanOpenSSLVersion}"
-								sh "conan export-pkg ../test/RESTAPICoreTestUtilities/conanfile.py RESTAPICoreTestUtilities/${version}@systelab/${channel} -s build_type=${props.conanBuildType} -s compiler.toolset=${props.conanCompilerToolset} -s arch=${props.conanArch} -o gtest=${props.conanGTestVersion} -o openssl=${props.conanOpenSSLVersion} --force"
-								sh "conan test ../test/RESTAPICoreTestUtilities/test_package/conanfile.py RESTAPICoreTestUtilities/${version}@systelab/${channel} -s build_type=${props.conanBuildType} -s compiler.toolset=${props.conanCompilerToolset} -s arch=${props.conanArch} -o gtest=${props.conanGTestVersion} -o openssl=${props.conanOpenSSLVersion}"
-								sh "bin/${props.conanBuildType}/RESTAPICoreTest.exe --gtest_output=xml:RESTAPICoreTest.xml"
+					archs.each
+						{ arch ->
+							configs.each
+							{ config ->
+								stage("Build ${config}|${arch}")
+								{
+									def buildFolder = "build/${config}-${arch}"
+									bat "conan install . --install-folder ${buildFolder} --profile=${profile} -s arch=${arch} -s build_type=${config}"
+									bat "conan build . --build-folder ${buildFolder}" 
+									bat "conan export-pkg . RESTAPICore/${version}@systelab/${channel} --build-folder ${buildFolder} --force"
+									bat "conan test ./test_package/conanfile.py RESTAPICore/${version}@systelab/${channel} --profile=${profile} -s arch=${arch} -s build_type=${config}"
+									bat "conan export-pkg ./test/RESTAPICoreTestUtilities/conanfile.py RESTAPICoreTestUtilities/${version}@systelab/${channel} --build-folder ${buildFolder} --force"
```

Comment:
Are test utilities build with first build? I think not being a different conanfile

Reply:
opsie... I'll change it, thanks

Reply:
It's a bit weird, but they are being build by the same makefile. They share the same dependencies. I know that this is a bit fucked up and we should fix it, but we should take a look at all the solutions here for the same problem.

Do you think we can do it in a next iteration?

Reply:
Sure, let's not delay anything because of this